docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,585 @@
+      "parent": "customp1",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    },,{
+      "perm_name": "customp3",
+      "parent": "active",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    },{
+      "perm_name": "owner",
+      "parent": "",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    }
+  ],
+  "total_resources": null,
+  "self_delegated_bandwidth": null,
+  "refund_request": null,
+  "voter_info": null,
+  "rex_info": null
+}
+```
+
+[[info]]
+| For simplicity we are using the same public private key pair for each permission, for production systems we recommend that you create and use new key pairs for each permission.
+
+## Link the Custom Permissions
+Once you have a custom permission you can link this custom permsssion to a smart contract action, requiring that permission level authorization, or higher, to execute the action.
+
+Link the custom permission _customp1_, to the _what_ action using the command [cleos set action permission](https://developers.eos.io/manuals/eos/latest/cleos/command-reference/set/set-action-permission.md):
+
+```shell
+cleos set action permission bob scholder what customp1 -p bob@active
+```
+
+The output should be similar to:
+
+```console
+executed transaction: 64198d1cc5f7dedf8809b86f22801eb004d50365ba72f8e2833ed191c6f6e30b  128 bytes  471 us
+#         eosio <= eosio::linkauth              {"account":"bob","code":"scholder","type":"what","requirement":"customp1"}
+```
+
+Link the custom permission _customp2_, to the _how_ action using the command `cleos set action permission` (link)`:
+
+```shell
+cleos set action permission bob scholder how customp2 -p bob@active
+```
+
+The output should be similar to:
+
+```console
+executed transaction: 64198d1cc5f7dedf8809b86f22801eb004d50365ba72f8e2833ed191c6f6e30b  128 bytes  471 us
+#         eosio <= eosio::linkauth              {"account":"bob","code":"scholder","type":"how","requirement":"customp2"}
+```
+
+### Test It
+We now have linked two custom permissions, _customp1_ and _customp2_, to two actions, _what_ and _how_.
+
+* `active` is able to call _why_, _what_ and _how_. The permission `active` is the parent of _customp1_ so is able to call anything that _customp1_ can call.
+* _customp1_ is able to call _what_ and _how_.  The permission _customp1_ is the parent of _customp2_ so is able to call anything that _customp2_ can call.
```

Comment:
For consistency with previous instances:
\_what\_ and \_how\_ -> \`what\` and \`how\`